pkg/apis/proxy/register.go

```diff
@@ -0,0 +1,70 @@
+package proxy
+
+import (
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"github.com/redhat-appstudio/sprayproxy/pkg/apis/proxy/v1alpha1"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+func (p *SprayProxy) GetBackends(c *gin.Context) {
+	c.String(http.StatusOK, "Backend urls: ", p.backends)
+}
+
+func (p *SprayProxy) RegisterBackend(c *gin.Context) {
+	zapCommonFields := []zapcore.Field{
+		zap.String("method", c.Request.Method),
+		zap.String("path", c.Request.URL.Path),
+		zap.String("query", c.Request.URL.RawQuery),
+		zap.Bool("dynamic-backends", p.enableDynamicBackends),
+	}
+	var newUrl v1alpha1.Backend
+	if err := c.ShouldBindJSON(&newUrl); err != nil {
+		c.String(http.StatusBadRequest, "please provide a valid json body")
+		p.logger.Info("backend server register request to proxy is rejected, invalid json body", zapCommonFields...)
+		return
+	}
+	zapBackendFields := append(zapCommonFields, zap.String("backend", newUrl.URL))
```

Comment:
You do not need a new variable.
```suggestion
	zapCommonFields := append(zapCommonFields, zap.String("backend", newUrl.URL))
```